src/cmd/edit.rs

```diff
@@ -0,0 +1,267 @@
+                ValidationResult::Retry => continue,
+            }
+        }
+        println!("Zoxide database not altered");
+        Ok(())
+    }
+}
+
+fn get_db_edits() -> Result<Option<String>> {
+    let data_dir = config::data_dir()?;
+    let mut db = DatabaseFile::new(data_dir);
+    let mut db = db.open()?;
+    let mut stream = db.stream(util::current_time().unwrap());
+    let mut to_edit = String::from(HEADER);
+    while let Some(dir) = stream.next() {
+        writeln!(&mut to_edit, "{},{},{}", dir.last_accessed, dir.rank, dir.path)?;
+    }
+    let edit_result = edit(&to_edit)?;
+    if edit_result == to_edit {
+        // The file was not changed at all so we don't want to attempt to overwrite the original
+        Ok(None)
+    } else {
+        Ok(Some(edit_result))
+    }
+}
+
+fn get_problems(db: &mut Database, db_edits: String) -> Problems {
+    let lines = db_edits.lines();
+
+    let mut errors: Vec<Problem> = Vec::new();
+    let mut warnings: Vec<Problem> = Vec::new();
+
+    for (index, line) in lines.enumerate() {
+        let line_number = index + 1;
+        let first_char = line.trim().chars().next();
+        if let Some(first_char) = first_char {
+            if first_char == '#' {
+                continue;
+            }
+        } else {
+            continue;
+        }
+        let mut split = line.split(',');
+        let (last_accessed_txt, rank_txt, path_txt) = (split.next(), split.next(), split.next());
+        if split.next().is_some() {
+            errors.push(Problem::new(line_number, "too many values on line".to_string()));
+            continue;
+        }
+
+        let last_accessed: Option<Epoch> = match last_accessed_txt {
+            Some(value) => match value.trim().parse::<Epoch>() {
+                Ok(value) => Some(value),
+                Err(e) => {
+                    errors.push(Problem::new(line_number, e.to_string()));
+                    None
+                }
+            },
+            None => {
+                errors.push(Problem::new(line_number, "Cannot parse 'last_accessed' field".to_string()));
+                None
+            }
+        };
+
+        let rank: Option<Rank> = match rank_txt {
+            Some(value) => match value.trim().parse::<Rank>() {
+                Ok(value) => Some(value),
+                Err(e) => {
+                    errors.push(Problem::new(line_number, e.to_string()));
+                    None
+                }
+            },
+            None => {
+                errors.push(Problem::new(line_number, "Cannot parse 'rank' field".to_string()));
+                None
+            }
+        };
+
+        let path: Option<String> = match path_txt {
+            Some(value) => {
+                if value.trim() != value {
+                    warnings.push(Problem::new(line_number, "path contains trailing whitespace".to_string()));
+                }
+                match resolve_path(&PathBuf::from(value)) {
```

Comment:
Let's not use `resolve_path` here. The user is directly editing the database, and is therefore expected to put in valid paths. If they're not valid, they'll just never show up in the results anyway. I'd say just test with `Path::is_absolute`.